pkg/resource/db_instance/hooks.go

```diff
@@ -160,3 +177,826 @@ func instanceDeleting(r *resource) bool {
 	dbis := *r.ko.Status.DBInstanceStatus
 	return dbis == StatusDeleting
 }
+
+// function to create restoreDbInstanceFromDbSnapshot payload and call restoreDbInstanceFromDbSnapshot API
+func (rm *resourceManager) restoreDbInstanceFromDbSnapshot(
+	ctx context.Context,
+	r *resource,
+) (created *resource, err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.restoreDbInstanceFromDbSnapshot")
+	defer func(err error) { exit(err) }(err)
+
+	input, err := rm.restoreDbInstanceFromDbSnapshotPayload(r)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, respErr := rm.sdkapi.RestoreDBInstanceFromDBSnapshotWithContext(ctx, input)
+
+	rm.metrics.RecordAPICall("CREATE", "RestoreDbInstanceFromDbSnapshot", respErr)
+	if respErr != nil {
+		return nil, respErr
+	}
+
+	// Merge in the information we read from the API call above to the copy of
+	// the original Kubernetes object we passed to the function
+	ko := r.ko.DeepCopy()
+
+	if resp.DBInstance.ActivityStreamEngineNativeAuditFieldsIncluded != nil {
```

Comment:
I will message you about how we can make the code-generator generate all of this ugly code

Reply:
Thanks Nick!